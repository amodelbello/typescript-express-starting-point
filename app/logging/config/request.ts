import dotenv from 'dotenv'
import * as colors from '../colors'
import { TransformableInfo } from 'logform'
import { format, transports } from 'winston'
dotenv.config()
const { combine, timestamp, label, printf } = format

const config = {
  level: process.env.LOG_LEVEL /* istanbul ignore next */ || 'info',
  format: combine(
    format(
      (info): TransformableInfo => {
        info.level = info.level.toLocaleUpperCase()
        return info
      },
    )(),
    combine(
      label({
        label: 'Request',
      }),
      timestamp({
        format: 'YYYY-MM-DD h:mm:SS A',
      }),
      printf(({ label, timestamp, level, message }): string => {
        return `${timestamp} [${colors.gray(label)}] ${colors.green(level)} : ${colors.gray(message)}`
      }),
    ),
  ),
  transports: [
    new transports.Console({
      silent: process.env.NODE_ENV === 'testing' && process.env.LOG_OUTPUT_WHILE_RUNNING_TESTS !== 'true',
    }),
  ],
}
export default config
