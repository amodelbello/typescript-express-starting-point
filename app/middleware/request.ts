import { Request, Response, NextFunction } from 'express'
import { v4 as uuidv4 } from 'uuid'

export const addFingerprintToRequest = (req: Request, res: Response, next: NextFunction): void => {
  req.id = uuidv4()
  next()
}
