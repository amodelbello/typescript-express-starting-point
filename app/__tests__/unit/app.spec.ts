import request from 'supertest'
import ExpressServer from '../../app'

const server = new ExpressServer(4002)
const { app } = server

beforeEach(async () => {
  await server.run(true)
})

afterEach(async () => {
  await server.stop()
})

describe('App', () => {
  it('Runs from `bin/run.ts`', async () => {
    const serverScript = await require('../../bin/run').server
    await request(serverScript.app)
      .get('/')
      .expect(200)
      .then(({ body }) => {
        expect(body).toMatchSnapshot()
      })
    await serverScript.stop()
  })

  it('Sends a request to /`', async () => {
    await request(app)
      .get('/')
      .expect(200)
      .then(({ body }) => {
        expect(body).toMatchSnapshot()
      })
  })

  it('Sends a request to /error`', async () => {
    await request(app)
      .get('/error')
      .expect(500)
      .then(({ body }) => {
        expect(body).toMatchSnapshot()
      })
  })
})
