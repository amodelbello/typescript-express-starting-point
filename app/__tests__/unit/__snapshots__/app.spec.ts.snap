// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Runs from \`bin/run.ts\` 1`] = `
Object {
  "hello": "Hello from the server",
}
`;

exports[`App Sends a request to /\` 1`] = `
Object {
  "hello": "Hello from the server",
}
`;

exports[`App Sends a request to /error\` 1`] = `Object {}`;
